### DISCUSSION TITLE
[further] development

### DISCUSSION THREAD

**Discussion by VW181-Cologne** [wxFormBuilder/wxFormBuilder#839] [General]
Hi community, Hi developer.

I would like to contribute my ideas to the wxFormBuilder.
But this without embarrassing me. 
So first try before I communicate the ideas to the world.

For this I would like to build on my computer a development environment for testing.
(MSYS2 produces only errors)

I would like to use Code::Blocks as IDE, but do not know how to get the source code integrated so that I can edit it further. 
If you could help me with your expertise.
(Probably just an adaptation of the makefiles)

This Text is a German-Translation.

greetings

Jörg

**Comment by sodevel** [comment]
Windows has no standard directory layout like Linux which makes software development more difficult when external libraries are involved. For wxFB using MSYS2 is much more simpler than using a more native approach because MSYS2 already contains wxWidgets and Boost as compiled libraries (however wxFB currently doesn't need compiled components of Boost). Only starting your IDE through MSYS2 might be a little cumbersome (but i don't know for sure because i am not using MSYS2 for IDE driven development).

I would suggest you try MSYS2 first, the instructions in the README file should be current, so i wonder what errors do you get?

wxFB uses CMake as its sole build system (generator), i highly suggest to use an IDE that has CMake integration because this simplifies the development a lot. This does exclude CodeBlocks which aged quite bad. I am using VS Code together with the Visual Studio Compiler. To use VS Code for wxFB inside MSYS2 you need to setup MSYS2 like outlined in the README, install the VS Code Extensions for C++ Development, CMake Tools and highly recommended GitLens. Then you start VS Code from the MSYS2 shell, open your wxFB repository and everything should work out-of-the-box.

If you really don't want to use MSYS2 more manual setup is required. Basically you have to do three things: Compile the wxWidgets library (or acquire a compatible precompiled one) and get the Boost source code, make CMake find wxWidgets and Boost, make wxFB find the wxWidgets DLLs during runtime. For the last two steps there are multiple options, i am using a scripted approach (which does much more and is too much to share) to define the environment variable `CMAKE_PREFIX_PATH` and point it to the libraries and modify the environment variable `PATH` so that the DLLs are found.
